servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/RetryStrategies.java

```diff
@@ -58,10 +58,11 @@ public static BiIntFunction<Throwable, Completable> retryWithConstantBackoffFull
         checkMaxRetries(maxRetries);
         requireNonNull(timerExecutor);
         requireNonNull(causeFilter);
-        final long delayNanos = delay.toNanos();
-        checkFullJitter(delayNanos);
+        final long jitterNanos = delay.toNanos();
+        checkFullJitter(jitterNanos);
+        final long upperBound = jitterNanos + 1; // add 1 because upper bound is non-inclusive.
         return (retryCount, cause) -> retryCount <= maxRetries && causeFilter.test(cause) ?
-                timerExecutor.timer(current().nextLong(0, delayNanos), NANOSECONDS) : failed(cause);
+                timerExecutor.timer(current().nextLong(0, upperBound), NANOSECONDS) : failed(cause);
```

Comment:
It feels like for the "full" jitter, 0 should either be an illegal value that throws or we can immediately jump to `failed(cause)` instead of waiting for 0 ns

Reply:
That is what the `checkFullJitter(..)` method ensures. While it may not matter in real life if we want to be technically correct we still need to add 1 to the jitterNanos or else by virtue of the upper bound being non-inclusive in `.nextLong(..)` we don't allow the full jitter but only up to `delay - 1ns`.

Reply:
And I realize I wasn't clear about that in the commit message. I can change it.